RELEASING.md

```diff
@@ -1,3 +1,8 @@
+IMPORTANT: Make sure you are using a JDK 8 JVM by checking `java -version` before any
+of the steps below.  If you run the steps below on a JDK 11+ JVM, that will break Java
+8 support, as the released jars will only run on JDK 11.  We do not target Java 8 when
+building on JDK 11 since Error Prone has required Java 11 since version 2.11.0.
```

Comment:
I am a bit worried about this part. Since our default environment for releasing uses JDK 11 as the default `java`. I can definitely switch JDKs for releasing, but I wonder if we can build in more of safety net here than just a comment in the `RELEASING.md` instructions.

Reply:
Any way to add a check that aborts the `publish` target if called with JDK 11?

Reply:
Good call!  I added a check in [c5feefc](https://github.com/uber/NullAway/pull/567/commits/c5feefc8398060ac4c2352ac4a426a5dc8e6a778).  I tested locally and the build failed when running `./gradlew publish` on JDK 11.  We'll have to observe and see that the CI job to publish a snapshot (on JDK 8) still works.